### PR TITLE
[DCJ-235] Fully populate LC on users for  /api/user/role/{roleName} User endpoint

### PR DESCRIPTION
**Pull request by aarohinadkarni**
### Addresses
https://broadworkbench.atlassian.net/browse/DCJ-235

### Summary
Fully populates the library card object & associated properties on a user now in the `getUsersAsRole` method exposed through the `/api/user/role/{roleName}` endpoint. Modifies existing tests to check for Library Cards.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: src/main/java/org/broadinstitute/consent/http/service/UserService.java [hunk 1]
```diff
@@ -456,6 +460,15 @@ private void addExistingLibraryCards(User user) {
     userDAO.updateUser(user.getDisplayName(), user.getUserId(), user.getInstitutionId());
   }
 
+  private void populateLibraryCardsOnUsers(List<User> users) {
+    for (User user: users) {
+      List<LibraryCard> cards = libraryCardDAO.findLibraryCardsByUserId(user.getUserId());
+      if (Objects.nonNull(cards) && !cards.isEmpty()) {
+        user.setLibraryCards(cards);
+      }
+    }
```
Comment (rushtong) [inline src/main/java/org/broadinstitute/consent/http/service/UserService.java]: This is a flag for me when I see a DAO call in a loop. Instead of 1 database call for users with LCs, we're now making N+1 database calls. Ideally, we should be doing the population in `userDAO.getUsersFromInstitutionWithCards()` and `userDAO.findUsersWithLCsAndInstitution()`. Those methods should be fully populating the DAA associations as part of the library card - their names suggest as much. They are a bit thorny, but I think it's worth it to refactor them into text block queries and ditch the query field construction pattern for both of them.

File: src/main/java/org/broadinstitute/consent/http/service/UserService.java [hunk 2]
```diff
@@ -269,13 +269,17 @@ public List<User> getUsersAsRole(User user, String roleName) {
...
+          return users;
         } else {
           throw new NotFoundException("Signing Official (user: " + user.getDisplayName()
               + ") is not associated with an Institution.");
         }
       case Resource.ADMIN:
-        return userDAO.findUsersWithLCsAndInstitution();
+        List<User> users = userDAO.findUsersWithLCsAndInstitution();
+        populateLibraryCardsOnUsers(users);
+        return users;
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/consent/http/service/UserService.java]: We should add a default case too, even if it does nothing.

File: src/main/java/org/broadinstitute/consent/http/service/UserService.java [hunk 3]
```diff
@@ -269,13 +269,17 @@ public List<User> getUsersAsRole(User user, String roleName) {
...
     }
     return Collections.emptyList();
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/consent/http/service/UserService.java:282]: Could we add a test for this line?

File: src/main/java/org/broadinstitute/consent/http/service/UserService.java [hunk 4]
```diff
@@ -456,6 +460,15 @@ private void addExistingLibraryCards(User user) {
     userDAO.updateUser(user.getDisplayName(), user.getUserId(), user.getInstitutionId());
   }
 
+  private void populateLibraryCardsOnUsers(List<User> users) {
+    for (User user: users) {
+      List<LibraryCard> cards = libraryCardDAO.findLibraryCardsByUserId(user.getUserId());
+      if (Objects.nonNull(cards) && !cards.isEmpty()) {
```
Comment (fboulnois) [inline src/main/java/org/broadinstitute/consent/http/service/UserService.java]: We should also test the null case here.

**Review by fboulnois — APPROVED** [review]
I agree with Greg's comment, but also here are a few comments relating to tests:

**Review by rushtong — APPROVED** [review]
Thank you for the changes! 👍🏽
